rasa/core/events/__init__.py

```diff
@@ -660,59 +686,106 @@ def as_dict(self) -> Dict[Text, Any]:
+
         return [
             ReminderScheduled(
-                parameters.get("action"),
+                parameters.get("intent"),
                 trigger_date_time,
-                parameters.get("name", None),
-                parameters.get("kill_on_user_msg", True),
-                parameters.get("timestamp"),
-                parameters.get("metadata"),
+                parameters.get("entities"),
+                name=parameters.get("name"),
+                kill_on_user_message=parameters.get("kill_on_user_msg", True),
+                timestamp=parameters.get("timestamp"),
+                metadata=parameters.get("metadata"),
             )
         ]
 
 
 # noinspection PyProtectedMember
 class ReminderCancelled(Event):
-    """Cancel all jobs with a specific name."""
+    """Cancel certain jobs."""
 
     type_name = "cancel_reminder"
 
     def __init__(
         self,
-        action_name: Text,
+        name: Optional[Text] = None,
+        intent: Optional[Text] = None,
+        entities: Optional[List[Dict]] = None,
         timestamp: Optional[float] = None,
         metadata: Optional[Dict[Text, Any]] = None,
     ):
-        """
+        """Creates a ReminderCancelled event.
+
+        All arguments are optional and used to refine the set of reminders that
+        are to be cancelled. If no arguments are supplied, this will cancel all reminders.
+
         Args:
-            action_name: name of the scheduled action to be cancelled
-            metadata: optional event metadata
+            name: Name of the reminder to be cancelled.
+            intent: Intent name that is to be used to identify the reminders to be cancelled.
+            entities: Entities that are to be used to identify the reminders to be cancelled.
+            timestamp: Optional timestamp.
+            metadata: Optional event metadata.
         """
 
-        self.action_name = action_name
+        self.name = name
+        self.intent = intent
+        self.entities = entities
         super().__init__(timestamp, metadata)
 
     def __hash__(self) -> int:
-        return hash(self.action_name)
+        return hash((self.name, self.intent, self.entities,))
 
     def __eq__(self, other) -> bool:
         return isinstance(other, ReminderCancelled)
 
     def __str__(self) -> Text:
-        return f"ReminderCancelled(action: {self.action_name})"
+        return f"ReminderCancelled(name: {self.name}, intent: {self.intent}, entities: {self.entities})"
+
+    def cancels_job_with_name(self, job_name: Text, sender_id: Text) -> bool:
+        """Determines if this ReminderCancelled should cancel the job with the given name.
+
+        Args:
+            job_name: Name of the job to be tested.
+            sender_id: The `sender_id` of the tracker.
+
+        Returns:
+            `True`, if this `ReminderCancelled` should cancel the job with the given name,
+            and `False` otherwise.
+        """
+        match = re.match(
```

Comment:
could you not just comma separate things and then split by `,` instead of having this fancy (and complicated) regex?

Reply:
I have no control over how `ACTION_NAME_SENDER_ID_CONNECTOR_STR` or `sender_id` look like. So I thought we should be as specific as possible here, to avoid conflicts. What, for example, if `sender_id` contains a comma?

Reply:
> What, for example, if sender_id contains a comma?

but it also could contain sth which breaks your regex right?

Reply:
I now added `re.escape` to make this impossible.

Reply:
... and I hash the name/intent/entities, so only digits and `-` can appear.